### PR TITLE
Emacs and CMakeLists.txt administrivia

### PR DESCRIPTION
**Pull request by rocky**
Some small little developer-like changes
* `.editorconfig` - specify `CMakeLists.txt` indentation
* `.gitignore`
   - ignore emacs backups 
   - `tmp` - like `/tmp` but specific to this project
   - `build*` - allow multiple kinds of build directories

### Checklist
- [ ] Code compiles correctly
- [ ] All tests are passing
- [ ] New tests have been created which fail without the change (if possible)
- [ ] README / documentation was extended, if necessary
- [ ] Changelog entry (if change is visible to the user)
- [ ] Used meaningful commit messages

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -44,6 +45,9 @@ deps/cache
 [._]*.sw[a-p]
 [._]sw[a-p]
 
+# emacs stuff
```
Comment (leonardoalt) [inline .gitignore:48]: also `vim`

Comment (leonardoalt) [reply .gitignore:48]: I guess this could be part of the IDE section

Comment (rocky) [reply .gitignore:48]: I don't understand what you mean - please elaborate. Or better just commit on top. I mostly care that ignorable files are ignored. What section they are in or additional comments are great, but I don't have a feel for that. 

Right now there is: 

```
# vim stuff
[._]*.sw[a-p]
[._]sw[a-p]

# emacs stuff
*~

# IDE files
...
```
What would you have instead?

Comment (leonardoalt) [reply .gitignore:48]: Oh ok, missed the vim section. I meant that vim might also save `*~` files, so the line you added is also related to vim. But yea, not too important

File: .editorconfig [hunk 2]
```diff
@@ -17,3 +17,7 @@ indent_size = 4
 [std/**.sol]
 indent_style = space
 indent_size = 4
+
+[*.{txt,cmake}]
```
Comment (leonardoalt) [inline .editorconfig:21]: Does this propagate to subdirectories?

Comment (rocky) [reply .editorconfig:21]: Yes. 

I couldn't figure out however to make it only apply to `CMakeList.txt` though, so it applies to all `.txt` files which probably doesn't do damage.

Comment (leonardoalt) [reply .editorconfig:21]: Ok!

**Review by leonardoalt — APPROVED** [review]

**Review by ekpyron — APPROVED** [review]

**Pull request merged**